lib/matplotlib/widgets.py

```diff
@@ -330,6 +334,11 @@ def __init__(self, ax, label, valmin, valmax, valinit=0.5, valfmt=None,
         self.valmin = valmin
         self.valmax = valmax
         self.valstep = valstep
+        self.valsnap = valsnap
+        if self.valsnap is not None:
```

Comment:
This isn't a great pattern because now if `valsnap is None`, we silently drop the user's input for `valstep` on the floor.  This is the sort of thing that causes the greatest anger at libraries, you change the thing AND NOTHING HAPPENS.

We should find a way to either fold `valstep` and `valsnap` into one parameter or raise if both are passed.

Folding them into one parameter seems ok (there is a bit of ambiguity else where about plural) and an the code shows, you can generate the snap location from a step.   In other cases where we have colliding kwargs we raise (e.g. norm vs density in hist).

Reply:
Ah indeed. Are you expressing a preference for a single kwarg here? I'm 55/45 towards two kwargs (appropriately ordered) and raising an error if both are passed. I think there is also an opportunity for user anger when they discover that a feature exists but was not very discoverable.

Reply:
I just pushed a change so it is two kwargs and raises an error if both are given. If need be I'm happy to move to a single `valstep` kwarg

Reply:
You can just do this above,  before the assignment, as `if valstep is not None and valsnap is not None:...`

Reply:
I moved the `ValueError` up to be with the other `ValueError`s